app/index.html

```diff
@@ -90,9 +90,37 @@
             <div class="col-md-6"><pre ng-show="user.period">{{user.period}}</pre></div>
           </div>
 
-          <p>Form valid: {{userForm.$valid}}</p>
+          <div class="form-group" field-state="user.food">
+            <label class="col-md-2 control-label">Favorite food</label>
+            <div class="col-md-4" af-field checkboxes-required name="user.food" ng-model="user.food">
+              <div class="checkbox" ng-repeat="foodItem in food">
+                <label>
+                  <input type="checkbox" af-field ng-model="user.food[foodItem]" ng-value="foodItem">
+                  {{foodItem}}
+                </label>
+              </div>
+              <div af-message="user.food"></div>
+            </div>
+            <div class="col-md-6"><pre ng-show="user.food">{{user.food}}</pre></div>
+          </div>
+        </fieldset>
 
+        <fieldset>
+          <legend>Friends</legend>
+          <button type="button" class="btn btn-default" ng-click="newFriend()"><i class="glyphicon glyphicon-plus"></i> New friend</button>
+          <div ng-form name="friendsForm" ng-repeat="friend in friends">
```

Comment:
Op zich logisch, want hij moet uniek zijn. In theorie kun je een error message hebben die buiten de repeater staat die de error voor het derde veld moet tonen ofzo.

Je kunt ook intern het messageId prefixen met de form name en $index in de form name gebruiken. Dan is het wat meer consistent met hoe angular het doet. Maar dan maak je je wel weer afhankelijk van (ng)form. Het enige van je kunt met ngForm is de `name` van je velden niet indexen zodat je een veldctrl via `$scope.friendsForm0['friend.name']` kunt pakken.